gomap_scan.go

```diff
@@ -111,10 +114,44 @@ func scanIPPorts(hostname string, laddr string, proto string, fastscan bool, ste
 // scanPort scans a single ip port combo
 // This detection method only works on some types of services
 // but is a reasonable solution for this application
-func scanPort(resultChannel chan<- portResult, protocol, hostname, service string, port int) {
+func scanPort(resultChannel chan<- portResult, protocol, hostname, service string, port int, proxyURL string) {
 	result := portResult{Port: port, Service: service}
 	address := hostname + ":" + strconv.Itoa(port)
-	conn, err := net.DialTimeout(protocol, address, 3*time.Second)
+	var conn net.Conn
+	var err error
+	var proxyDialer proxy.Dialer
+	// use proxy dialer if proxyURL string is not empty
+	if len(proxyURL) > 0 {
```

Comment:
```
if proxyURL != nil {
```

Reply:
hi I'm checking for length of proxy string because string cannot be set to `nil` -- https://stackoverflow.com/a/18595217/5502033